upup/pkg/fi/cloudup/awstasks/subnet.go

```diff
@@ -98,6 +99,14 @@ func (e *Subnet) Find(c *fi.Context) (*Subnet, error) {
 		}
 
 		actual.IPv6CIDR = association.Ipv6CidrBlock
+		if e.IPv6CIDR != nil && strings.HasPrefix(aws.StringValue(e.IPv6CIDR), "/") {
+			subnetIPv6CIDR, err := calculateSubnetCIDR(e.VPC.IPv6CIDR, e.IPv6CIDR)
```

Comment:
`VPCAmazonIPv6CIDRBlock.RenderAWS()` cannot write the CIDR, because the request doesn't actually return which block it associates. Because of that, there is no dependency and `Subnet.RenderAWS()` waits for the association to be done, using `fi.NewTryAgainLaterError()`.

I agree that `Find()` writing to `e` is awkward, but it was that or writing the `e.IPv6CIDR` notation to `a.IPv6CIDR`.
Also makes it easier later in `FindDeletions()`. I could change that to set `a.IPv6CIDR = e.IPv6CIDR`, if it makes things less awkward.

Reply:
I think `Find()` writing to `e` is okay, I'm just having to figure out what the rules and model are. It seems `Find()` can refine the expected based on existing state, including state produced by tasks listed as dependencies. But it has to deal with those dependencies not running due to `TargetDryRun`.

There's an opportunity here to improve the developer documentation.

This could be used in the future to support a `/64` notation where the particular sub-block doesn't need to be specified.

But there is a dependency on `VPCAmazonIPv6CIDRBlock` as the association won't happen unless that task is run. And there shouldn't be two blocks of code when one would do, especially not if the two blocks of code implement different semantics.